projects/apache-commons-configuration/XMLConfigurationWriteFuzzer.java

```diff
@@ -0,0 +1,47 @@
+import com.code_intelligence.jazzer.api.FuzzedDataProvider;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringWriter;
+
+import org.apache.commons.configuration2.XMLConfiguration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.configuration2.io.ConfigurationLogger;
+import org.apache.commons.configuration2.io.FileHandler;
+
+public class XMLConfigurationWriteFuzzer {
+    public static void fuzzerTestOneInput(FuzzedDataProvider data) {
+        // Create needed objects
+        final File tempFile;
+        final String absoluteFilepath;
+        try {
+            tempFile = File.createTempFile("XMLConfiguration", "xml");
+            tempFile.deleteOnExit();
+
+            absoluteFilepath = tempFile.getAbsolutePath();
+
+            final FileWriter fileWriter = new FileWriter(tempFile);
+            fileWriter.write(data.consumeRemainingAsString());
+            fileWriter.close();
+        } catch (IOException ioe) {
+            // Preparations failed ; exit early
+            return;
+        }
+
+        final XMLConfiguration xmlConfig = new XMLConfiguration();
+        xmlConfig.setLogger(null); // disable the logger
+
+        final FileHandler fileHandler = new FileHandler(xmlConfig);
+        fileHandler.setPath(absoluteFilepath);
+
+        final StringWriter writer = new StringWriter();
+
+        try {
+            fileHandler.load();
+            xmlConfig.write(writer);
```

Comment:
Shouldn't you also add fuzz data in the `StringWriter`?

Reply:
I don't think so.
My idea was to see if the parser could read the data but show bugs while writing it back.
I am not sure if adding data to the writer before or after calling `xmlConfig.write` would change anything.

Reply:
Alright